### PR TITLE
ios purchase handling

### PR DESCRIPTION
**Pull request by nasimafeef**

### CODE REVIEW

File: js/browser.js [hunk 1]
```diff
@@ -66,6 +70,20 @@ var Billing = Class(Emitter, function (supr) {
...
+    }
+    return false;
+  }
+
+  this.isNotiOSDevice = function () {
+    var device_type =  util.getDeviceInfo().type;
+    return (device_type != 'IPhone'
+      && device_type != 'IPad'
+      && device_type != 'IPod');
+  };
```
Comment (rampr) [inline js/browser.js]: as discussed, this check is not needed

File: js/browser.js [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
+    if (is_not_iOS || this.isPaymentsReady()) {
       fbinstant.purchaseAsync(product)
         .then(bind(this, function (purchase) {
           this.afterPurchase(purchase, access_token);
         }))
         .catch(bind(this, function (e) {
           this.onFailure(product_id);
         }));
+    } else if (!is_not_iOS) {
+        this.onFailure(product_id, true);
```
Comment (rampr) [inline js/browser.js]: indentation

File: js/browser.js [hunk 3]
```diff
@@ -1,31 +1,35 @@
...
     } else {
-      this.onFailure(product_id);
+        this.onFailure(product_id);
```
Comment (rampr) [inline js/browser.js]: indenation - should be 2 spaces

File: js/browser.js [hunk 4]
```diff
@@ -1,31 +1,35 @@
...
 
     if (payload) {
       product.developerPayload = payload;
     }
 
-    if (GC.app.payments_ready) {
+    if (is_not_iOS || this.isPaymentsReady()) {
       fbinstant.purchaseAsync(product)
         .then(bind(this, function (purchase) {
           this.afterPurchase(purchase, access_token);
```
Comment (rampr) [inline js/browser.js]: we can call this.consumeItem directly

File: js/browser.js [hunk 5]
```diff
@@ -66,6 +70,20 @@ var Billing = Class(Emitter, function (supr) {
     }
   };
 
+  this.isPaymentsReady = function () {
+    if (GC.app.payments_ready) {
+      return true;
+    }
+    return false;
+  }
```
Comment (rampr) [inline js/browser.js]: best way would be to move this function to FBInstant plugin itself. That way we don't have to store state in `GC.app.payments_ready` and also we don't have to pass the state around

File: js/browser.js [hunk 6]
```diff
@@ -1,31 +1,35 @@
 import event.Emitter as Emitter;
 import fbinstant as fbinstant;
+import quest.modules.util as util;
```
Comment (vivekb91) [inline js/browser.js]: is this needed?

**Comment by vivekb91** [comment]
@nasimafeef is there any option for localisation of currency?

**Review by rampr — APPROVED** [review]

**Review by vivekb91 — APPROVED** [review]

**Pull request merged**